packages/xo-server/src/api/vm.js

```diff
@@ -356,8 +356,10 @@ async function delete_ ({
   )
 
   // Update resource sets
-  const resourceSet = xapi.xo.getData(vm._xapiId, 'resourceSet')
-  if (resourceSet != null) {
+  if (
+    vm.$snapshot_of === undefined &&
```

Comment:
Test `vm.is_a_snapshot` instead.

Reply:
`is_a_snapshot` does not exist on XO objects. I could simply test `vm.type` instead, though.

Reply:
This seems better because you could have false positives otherwise.